### PR TITLE
[EISW-89824] [master] Rename VPUX to NPU

### PR DESCRIPTION
**Pull request by StefaniaHergane**
### Details:
 - Initial PR https://github.com/openvinotoolkit/openvino/pull/18969
 - Similar PR for `releases/2023/0` branch: https://github.com/openvinotoolkit/openvino/pull/19002 

### Tickets:
 - EISW-89824

### CODE REVIEW

**Comment by ilya-lavrenov** [comment]
build_jenkins

File: src/tests/functional/plugin/shared/include/behavior/infer_request/config.hpp [hunk 1]
```diff
@@ -117,7 +117,7 @@ TEST_P(InferRequestConfigTest, ReusableCPUStreamsExecutor) {
         // Load CNNNetwork to target plugins
         execNet = ie->LoadNetwork(cnnNet, target_device, config);
         execNet.CreateInferRequest();
-        if (target_device == ov::test::utils::DEVICE_KEEMBAY) {
```
Comment (iefode) [inline src/tests/functional/plugin/shared/include/behavior/infer_request/config.hpp:120]: General comment, please do not hardcode such constructions in the shared test lib. TODO: remove

File: src/tests/test_utils/common_test_utils/include/common_test_utils/file_utils.hpp [hunk 2]
```diff
@@ -365,7 +365,7 @@ class MockPlugin : public ov::IPlugin {
 }  // namespace ov
 
 
-// vpu repo uses CommonTestUtils::
+// npu repo uses CommonTestUtils::
 // so we need to add these names to CommonTestUtils namespace
 namespace CommonTestUtils {
```
Comment (iefode) [inline src/tests/test_utils/common_test_utils/include/common_test_utils/file_utils.hpp:370]: @olpipi  Why this code is still here??

**Review by iefode — COMMENTED** [review]
I left only one comment related to NPU. Please check. Also you need to check https://github.com/intel-innersource/frameworks.ai.openvino.ci.product-configs which is a part of instrastruture. You need to update `conf_dldt_tests.py` for conformance in case

**Comment by ilya-lavrenov** [comment]
build_jenkins

File: src/inference/include/ie/npu/myriad_config.hpp [hunk 3]
```diff
@@ -60,7 +60,7 @@ INFERENCE_ENGINE_1_0_DEPRECATED DECLARE_VPU_CONFIG(MYRIAD_PCIE);
 INFERENCE_ENGINE_1_0_DEPRECATED DECLARE_VPU_CONFIG(MYRIAD_USB);
```
Comment (wangleis) [inline src/inference/include/ie/npu/myriad_config.hpp:60]: Please update device name from VPU to NPU.

File: src/tests/test_utils/common_test_utils/include/common_test_utils/test_common.hpp [hunk 4]
```diff
@@ -18,7 +18,7 @@ class TestsCommon;
 
 // Temporal WA for applications.ai.vpu-accelerators.vpux-plugin repo
```
Comment (wangleis) [inline src/tests/test_utils/common_test_utils/include/common_test_utils/test_common.hpp:19]: Please update device name in comments from VPU to NPU.

File: src/inference/include/ie/npu/myriad_config.hpp (hunk 3 above)
Comment (ilya-lavrenov) [reply src/inference/include/ie/npu/myriad_config.hpp:60]: old headers should be untouched

**Comment by ilya-lavrenov** [comment]
build_jenkins

File: src/inference/include/ie/npu/myriad_config.hpp [hunk 5]
```diff
@@ -21,7 +21,7 @@
 #    endif
```
Comment (ilya-lavrenov) [inline src/inference/include/ie/npu/myriad_config.hpp:21]: please, revert folder renaming.
it's old VPU (which is MYRIAD), not VPUX

File: src/inference/include/ie/npu/myriad_config.hpp [hunk 6]
```diff
@@ -21,7 +21,7 @@
...
 #endif
 
-#include "vpu_config.hpp"
+#include "npu_config.hpp"
```
Comment (ilya-lavrenov) [inline src/inference/include/ie/npu/myriad_config.hpp:24]: revert

**Review by ilya-lavrenov — CHANGES_REQUESTED** [review]

File: src/inference/include/ie/npu/myriad_config.hpp (hunk 3 above)
Comment (StefaniaHergane) [reply src/inference/include/ie/npu/myriad_config.hpp]: Done.

File: src/tests/test_utils/common_test_utils/include/common_test_utils/test_common.hpp (hunk 4 above)
Comment (StefaniaHergane) [reply src/tests/test_utils/common_test_utils/include/common_test_utils/test_common.hpp:19]: Done.

File: src/tests/test_utils/common_test_utils/include/common_test_utils/file_utils.hpp (hunk 2 above)
Comment (olpipi) [reply src/tests/test_utils/common_test_utils/include/common_test_utils/file_utils.hpp:370]: https://github.com/intel-innersource/applications.ai.vpu-accelerators.vpux-plugin/pull/5374

**Pull request merged**